src/stateManager/ValidationService.ts

```diff
@@ -1,384 +1,494 @@
+        };
+    }
 
-            // try to fetch later timestamp from on-chain data
-            const onChainTimestamp =
-                await this.fetchOnChainTimestamp(previousBlock);
+    // ────────────────────── INTERNAL ACTION METHODS ─────────────────────
 
-            // Early return: Couldn't fetch or not better than current
-            if (!onChainTimestamp || onChainTimestamp <= previousTimestamp) {
-                const fraudProof =
-                    this.disputeHandler.proofManager.createIncorrectDataProof(
-                        signedBlock
-                    );
-                await this.disputeHandler.createDispute(blk.forkId, "0x00", 0, [
-                    fraudProof
-                ]);
-                return dispute(AgreementFlag.INCORRECT_DATA);
-            }
+    private queueForLater(blockConfirmation: BlockConfirmationStruct): void {
+        this.storage.queues.queueConfirmation(blockConfirmation);
+    }
 
-            // Update the previous block with the on-chain timestamp
-            previousBlock.onChainTimestamp = onChainTimestamp;
-            this.storage.blocks.setOnChainTimestamp(
-                previousBlock.forkId,
-                previousBlock.height,
-                onChainTimestamp
-            );
+    // ────────────────────── VALIDATION METHODS ─────────────────────
 
-            // Re-validate with updated timestamp
-            const isValidTimestamp =
-                blk.timestamp >= onChainTimestamp &&
-                blk.timestamp <= onChainTimestamp + this.timeCfg.p2pTime;
+    isChannelOpen(forkId: ForkId): boolean {
+        return forkId !== ZeroHash;
+    }
 
-            if (!isValidTimestamp) {
-                const fraudProof =
-                    this.disputeHandler.proofManager.createIncorrectDataProof(
-                        signedBlock
-                    );
-                await this.disputeHandler.createDispute(blk.forkId, "0x00", 0, [
-                    fraudProof
-                ]);
-                return dispute(AgreementFlag.INCORRECT_DATA);
+    private isLinked(block: Block): boolean {
+        const { forkId, height } = block.coordinates;
+        if (height === 0) {
+            const genesisSnapshot =
+                this.storage.stateSnapshots.getGenesisSnapshotDataByForkId(
+                    forkId
+                );
+            if (!genesisSnapshot) {
+                return false;
             }
-            // If valid, continue with rest of validation
-        }
 
-        // Subjective timestamp validation (requireS)
-        if (
-            blk.onChainTimestamp === undefined ||
-            blk.onChainTimestamp <= blk.timestamp
-        ) {
-            if (
-                Math.abs(Clock.getTimeInSeconds() - blk.timestamp) >=
-                this.timeCfg.agreementTime
-            )
-                return timestampTooOld();
+            return genesisSnapshot.hash === block.previousBlockHash;
```

Comment:
I think we went for the convention `(block 0).previousBlockHash = 0x00`, and `block.forkId = hash(genesisStateSnapshot.snapshotData)` which is always true - so block 0 `can not be not linked <=> is always linked`, since it already commited to the genesis by defining the `forkId`